functions/dialect.go

```diff
@@ -187,7 +192,14 @@ type dialectFile struct {
 
 type dialectTypeInfo struct {
 	SqlTypeName       string `yaml:"sql_type_name"`
-	SupportedAsColumn bool   `yaml:"supported_as_column"`
+	SupportedAsColumn *bool  `yaml:"supported_as_column"`
```

Comment:
can we start using the defaults pattern that we introduced in the matching patches? Example [here](https://github.com/substrait-io/substrait-go/commit/1a800d967cdceef1f29c3a6d7fdd06ac5d37fc30#diff-00e5053c917f74402204735708fe3fb7f91142e641d66f3c1749f8f09f33ccc7R191).